pkg/web/http.go

```diff
@@ -25,9 +25,10 @@ import (
 )
 
 type controller struct {
-	cfg  *config.Config
-	auth *auth.Auth
-	qs   *search.QueryServer
+	cfg    *config.Config
+	auth   *auth.Auth
+	qs     *search.QueryServer
+	fields *map[string][]string
```

Comment:
I see no reason here to pass a pointer to the map. Maps are already reference types.